Ubuntu-20/Dockerfile

```diff
@@ -25,42 +26,46 @@ ENV TZ=UTC
 
 ENV GCC_MAJOR_VERSION=10
 
-# Update the package list
-RUN apt-get update
-
-# Set the locale.
-RUN apt-get install --yes --no-install-recommends locales && \
-    sed -i '/en_US.UTF-8/s/^# //g' /etc/locale.gen && \
-    locale-gen
-ENV LANG en_US.UTF-8
-ENV LANGUAGE en_US:en
-ENV LC_ALL en_US.UTF-8
-
-# Upgrade all packages.
-RUN apt-get upgrade -y
-
-# Install the basics
-RUN apt-get install -y cryptsetup apt-transport-https apt-utils \
-      software-properties-common sudo wget && \
-    apt-get install --yes python3.9 python3-pip python3.9-venv && \
-    update-alternatives --install /usr/bin/python python /usr/bin/python3.9 1 && \
-    update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 1
-
+# Install and update the package list
 # Add mono's PPA.  The stuart docs direct people to install directly from the
 # mono project, so we'll do the same.
 # - https://github.com/tianocore/edk2-pytool-extensions/blob/master/docs/usability/using_extdep.md#a-note-on-nuget-on-linux
-RUN apt-get install -y gnupg ca-certificates && \
+RUN apt-get update && \
+    apt-get install --yes --no-install-recommends \
+        software-properties-common \
+        apt-utils \
+        cryptsetup \
+        apt-transport-https \
+        sudo \
+        wget \
+        build-essential \
+        uuid-dev \
+        git \
+        virtualenv \
+        device-tree-compiler \
+        mono-devel \
+        wget \
+        python3.9\
+        python3-pip \
+        python3.9-venv \
+        locales \
+        gnupg \
+        ca-certificates && \
     apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF && \
-    echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list
-
-# Install toolchains.
-RUN apt install -y \
-    g++-${GCC_MAJOR_VERSION} gcc-${GCC_MAJOR_VERSION} \
-    g++-${GCC_MAJOR_VERSION}-x86-64-linux-gnux32 gcc-${GCC_MAJOR_VERSION}-x86-64-linux-gnux32 \
-    g++-${GCC_MAJOR_VERSION}-aarch64-linux-gnu gcc-${GCC_MAJOR_VERSION}-aarch64-linux-gnu \
-    g++-${GCC_MAJOR_VERSION}-riscv64-linux-gnu gcc-${GCC_MAJOR_VERSION}-riscv64-linux-gnu \
-    g++-${GCC_MAJOR_VERSION}-arm-linux-gnueabi gcc-${GCC_MAJOR_VERSION}-arm-linux-gnueabi \
-    g++-${GCC_MAJOR_VERSION}-arm-linux-gnueabihf gcc-${GCC_MAJOR_VERSION}-arm-linux-gnueabihf && \
+    echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list &&\
```

Comment:
Not sure I like hiding the mono PPA in here.

The steps to install mono were a direct lift from stuart docs.  I think it's best to keep them that way so it's easy to update them as the upstream docs change.

Reply:
Ok, I'll keep them.